File: 04084.py
while True:
    a, b, c, d = map(int, input().split())
    if not a and not b and not c and not d:
        break

    cnt = 0
    while a != b or a != c or a != d:
        tmp = a
        a = abs(a - b)
        b = abs(b - c)
        c = abs(c - d)
        d = abs(d - tmp)
        cnt += 1

    print(cnt)

File: 04084.cpp
#include <iostream>
#include <cmath>
using namespace std;

int main() {
	while (1) {
		int a, b, c, d; cin >> a >> b >> c >> d;
		if (!a && !b && !c && !d) break;

		int cnt = 0;
		while (a != b || a != c || a != d) {
			int tmp = a;
			a = abs(a - b);
			b = abs(b - c);
			c = abs(c - d);
			d = abs(d - tmp);
			cnt++;
		}

		cout << cnt << endl;
	}
}
